/*
 * This file was automatically generated by EvoSuite
 * Wed Apr 18 18:02:18 GMT 2018
 */

package com.jfinal.plugin.activerecord;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.jfinal.plugin.activerecord.NullDataSource;
import java.io.File;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.io.MockFile;
import org.evosuite.runtime.mock.java.io.MockPrintStream;
import org.evosuite.runtime.mock.java.io.MockPrintWriter;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false, useJEE = true)
public class NullDataSource_ESTest extends NullDataSource_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      Class<Integer> class0 = Integer.class;
      // Undeclared exception!
      try { 
        nullDataSource0.isWrapperFor(class0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      Class<Object> class0 = Object.class;
      // Undeclared exception!
      try { 
        nullDataSource0.unwrap(class0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.getParentLogger();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.getConnection();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.getConnection("% )sqDKZY2w*[]FSsm", "% )sqDKZY2w*[]FSsm");
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.getLoginTimeout();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      File file0 = MockFile.createTempFile("x\".4hoH9RVj]U", "][<UUA?&$+}LmBMjo7");
      MockPrintStream mockPrintStream0 = new MockPrintStream(file0);
      MockPrintWriter mockPrintWriter0 = new MockPrintWriter(mockPrintStream0, false);
      // Undeclared exception!
      try { 
        nullDataSource0.setLogWriter(mockPrintWriter0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.getLogWriter();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }

  @Test(timeout = 4000)
  public void test8()  throws Throwable  {
      NullDataSource nullDataSource0 = new NullDataSource();
      // Undeclared exception!
      try { 
        nullDataSource0.setLoginTimeout((-1));
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Can not invoke the method of NullDataSource
         //
         verifyException("com.jfinal.plugin.activerecord.NullDataSource", e);
      }
  }
}
